src/editors/containers/VideoUploadEditor/index.test.jsx

```diff
@@ -0,0 +1,113 @@
+import { render, fireEvent, act } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import VideoUploadEditor, { VideoUploader } from '.';
+
+const mockDispatch = jest.fn();
+const mockOnUpload = jest.fn();
+
+jest.mock('react-redux', () => ({
+  useDispatch: () => mockDispatch,
+}));
+
+const defaultEditorProps = {
+  intl: {},
+};
+
+const defaultUploaderProps = {
+  onUpload: mockOnUpload,
+  errorMessage: '',
+  intl: {},
+};
+
+const renderEditorComponent = (props = defaultEditorProps) => render(
+  <IntlProvider locale="en">
+    <VideoUploadEditor {...props} />
+  </IntlProvider>,
+);
+
+const renderUploaderComponent = (props = defaultUploaderProps) => render(
+  <IntlProvider locale="en">
+    <VideoUploader {...props} />
+  </IntlProvider>,
+);
+
+describe('VideoUploadEditor', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders without errors', () => {
+    const { container } = renderEditorComponent();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders the input field for pasting the video ID or URL', () => {
+    const { getByPlaceholderText } = renderEditorComponent();
+    const input = getByPlaceholderText('Paste your video ID or URL');
+    expect(input).toBeInTheDocument();
+  });
+
+  it('updates the input field value when user types', () => {
+    const { getByPlaceholderText } = renderEditorComponent();
+    const input = getByPlaceholderText('Paste your video ID or URL');
+
+    fireEvent.change(input, { target: { value: 'test value' } });
+    expect(input.value).toBe('test value');
+  });
+
+  it('shows error message with unsupported files', async () => {
+    const { getByTestId, findByText } = renderEditorComponent();
+    const fileInput = getByTestId('fileInput');
+
+    const unsupportedFile = new File(['(⌐□_□)'], 'unsupported.avi', { type: 'video/avi' });
+    fireEvent.change(fileInput, { target: { files: [unsupportedFile] } });
+
+    const errorMsg = await findByText('Video must be an MP4 or MOV file');
+    expect(errorMsg).toBeInTheDocument();
+  });
+});
+
+describe('VideoUploader', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders without errors', () => {
+    const { container } = renderUploaderComponent();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders with an error message', () => {
+    const errorMessage = 'Video must be an MP4 or MOV file';
+    const { getByText } = renderUploaderComponent({ ...defaultUploaderProps, errorMessage });
+    expect(getByText(errorMessage)).toBeInTheDocument();
+  });
+
+  it('handles unsupported video file formats', () => {
+    const { getByTestId } = renderUploaderComponent();
+    const fileInput = getByTestId('fileInput');
+    const unsupportedFile = new File(['(⌐□_□)'], 'unsupported.avi', { type: 'video/avi' });
+
+    fireEvent.change(fileInput, { target: { files: [unsupportedFile] } });
+    expect(mockOnUpload).not.toHaveBeenCalled();
+  });
```

Comment:
@pkulkark Does that actually call the `onDrop` call back? I don't see the logic in the `VideoUploader` which would filter the files by type. I think it passes them to the `onUpload` callback, which does the filtering. The fact that this test is passing means that `.change` doesn't trigger the `onDrop`.

Reply:
You're right. It doesn't. It can't really find the element because react-dropzone hides it. I'll remove it since we are covering it in `VideoUploadEditor` above.

Reply:
Done.